### PR TITLE
add number_of_replicas tests

### PR DESCRIPTION
**Pull request by wanliAlex**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
we add tests for `number_of_replicas` 

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
not

* **Other information**:
this PR is related to https://github.com/marqo-ai/marqo/pull/391

### CODE REVIEW

**Review by pandu-k — CHANGES_REQUESTED** [review]
Can you do a simple test where you add a custom number of shards (like) 3, and ensure that number is returned when you get the index settings?

**Comment by wanliAlex** [comment]
image
unit test passed (with added custom `number_of_replicas` test)

File: tests/v0_tests/test_get_settings.py [hunk 1]
```diff
@@ -58,6 +60,7 @@ def test_custom_settings(self):
 
         self.assertIn('index_defaults', index_settings)
         self.assertIn('number_of_shards', index_settings)
+        self.assertIn('number_of_shards', index_settings)
```
Comment (pandu-k) [inline tests/v0_tests/test_get_settings.py]: is this meant to be "number_of_replicas"?

File: tests/v0_tests/test_index.py [hunk 2]
```diff
@@ -189,4 +192,14 @@ def run():
...
         assert run()
 
-
+    def test_create_custom_number_of_replicas(self):
+        intended_replicas = 5
+        settings = {
+            "number_of_replicas": intended_replicas
+        }
+        self.client.create_index(index_name=self.index_name_1, settings_dict = settings)
+        resp = requests.get(
```
Comment (pandu-k) [inline tests/v0_tests/test_index.py]: could you just get this from the Marqo [get index settings endpoint](https://docs.marqo.ai/0.0.15/API-Reference/settings/)? We don't want any interaction with OpenSearch in py-marqo/ py-marqo tests:

**Review by pandu-k — CHANGES_REQUESTED** [review]

File: tests/v0_tests/test_get_settings.py (hunk 1 above)
Comment (wanliAlex) [reply tests/v0_tests/test_get_settings.py]: updated

File: tests/v0_tests/test_index.py (hunk 2 above)
Comment (wanliAlex) [reply tests/v0_tests/test_index.py]: updated

**Comment by wanliAlex** [comment]
image

**Review by pandu-k — APPROVED** [review]

**Pull request merged**